### ISSUE TITLE
Job.emitDependency Build Error

### ISSUE THREAD

**Issue by andre-s-sousa** [vercel/next.js#56516]
### Link to the code that reproduces this issue

https://github.com/andre-s-sousa/nextjs-build-issue

### To Reproduce

1. Run `npm install`
2. Run `npm run build`

### Current vs. Expected behavior

**Expected Behaviour**
Application builds without any issues
**Current Behaviour**
![Screenshot 2023-10-06 at 12 21 53](https://github.com/vercel/next.js/assets/65219481/47e445d6-c9b3-49f4-84ce-1ce0d2decb27)

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000
Binaries:
  Node: 20.7.0
  npm: 10.1.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.5-canary.2
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

Same code behaved as expected in previous versions, **including 13.5.3**

**Comment by andre-s-sousa** [comment]
Still in v13.5.5-canary.8!

**Comment by andre-s-sousa** [comment]
Seems to have been fixed somewhere along the way, in 13.5.5-canary.17

**Comment by balazsorban44** [comment]
Closing as per the above. Please open a new issue if you are still seeing this.
